src/tribol/physics/Mortar.cpp

```diff
@@ -631,6 +641,565 @@ void ComputeSingleMortarJacobian( SurfaceContactElem& elem )
   return;
 }
 
+#ifdef TRIBOL_USE_ENZYME
+
+//------------------------------------------------------------------------------
+int ApplyNormalEnzyme( CouplingScheme* cs )
+{
+  auto planes_view = cs->get3DContactPlanes().view();
+  auto& lm_opts = cs->getEnforcementOptions().lm_implicit_options;
+  bool compute_jacobian = false;
+  if ( lm_opts.eval_mode == ImplicitEvalMode::MORTAR_RESIDUAL_JACOBIAN ||
+       lm_opts.eval_mode == ImplicitEvalMode::MORTAR_JACOBIAN ) {
+    if ( lm_opts.sparse_mode == SparseMode::MFEM_ELEMENT_DENSE ) {
+      cs->getMethodData()->reserveBlockJ(
+          { BlockSpace::NONMORTAR, BlockSpace::MORTAR, BlockSpace::LAGRANGE_MULTIPLIER }, planes_view.size() );
+      cs->createNormalJacobian();
```

Comment:
I think normal info and functionality should move off the coupling scheme.

Reply:
Agreed, this could use some refactoring.  A block Jacobian storage class would be a good addition (instead of hijacking `MethodData`).  We could store that on a method-specific class.  But let's defer that to a future PR.